### PR TITLE
Added General Info Page for Registration

### PR DESCRIPTION
**Pull request by B-Timok**
## Pull Request to Complete Issue #3 
This PR introduces a new General Info page that is part of the user registration process. It allows new users to input their general information, such as name, username, email, phone number, and date of birth.

## Changes Made
- **New Components:**
  - `GeneralInfo`: The main component for the general information form.

- **State Management:**
  - Utilized React's `useState` for handling form inputs including first name, last name, username, email, phone number, and date of birth.

- **Form Inputs:**
  - Implemented a reusable `TextField` component for various inputs with validation for email and phone formats.

- **Styling:**
  - Applied styles using `GeneralInfo.module.scss`. Added a Google Font called 'Lato' and changed the weight to match the Figma design better.

- **Material UI Integration:**
  - Used Material UI components (`Button`, `Box`, `Stack`, `Typography`) for a consistent and modern user interface.

## Screenshots
image

## Next Steps
- Backend API integration to submit the gathered information.

I appreciate any feedback!

### CODE REVIEW

**Review by Linkaloo — APPROVED** [review]
This will actually be used inside the signup page and is not by itself a page. It is just a component to be rendered on the first step of signing up. Though its not an issue to render it as its own page right now

File: client/src/pages/GeneralInfo/GeneralInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import styles from './GeneralInfo.module.scss';
+import { useState } from 'react';
+import { Button, Box, Stack, Typography } from '@mui/material';
```
Comment (tbmcallister) [inline client/src/pages/GeneralInfo/GeneralInfo.tsx:3]: Make sure to use our custom Button component here instead of the MUI one

**Review by Aligary — APPROVED** [review]
I have no additional comments. Looks good!

File: client/src/pages/GeneralInfo/GeneralInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import { TextField } from '@/components';
+
+function GeneralInfo(): JSX.Element {
+  // console.log('Rendering GeneralInfo...');
```
Comment (morenj) [inline client/src/pages/GeneralInfo/GeneralInfo.tsx:7]: I recommend removing commented code to keep everything as neat as possible.

**Review by morenj — APPROVED** [review]
Great job on this! Everything looks good on my end. I checked out your branch and was able to test the new page/component.

**Pull request merged**